File: src/app/models/EdgeApplication.js


// Define collection and schema for EdgeApplication
export interface EdgeApplication {
    name:
	type : string
    version:
	type : string
    image:
	type : string
    Gateway:
	type : Schema.Types.ObjectId
    Status:
 	type : String
#
    collection: 'edgeApplications'
}

File: iotonaspdotnet/Domain/EdgeApplication.cs
using iotonaspdotnet.Domain.Contracts;

namespace iotonaspdotnet.Domain;

public class EdgeApplication
{
    public Guid Id { get; set; } = Guid.NewGuid();

 public virtual long EdgeapplicationId { get; set; }
 public virtual string Name { get; set; }
 public virtual string Version { get; set; }
 public virtual string Image { get; set; }
public virtual Gateway Gateway { get; set; }
 public virtual DeploymentStatus Status { get; set; }

    public static EdgeApplication FromRequest(EdgeApplicationRequest request) {
        return new EdgeApplication {
            Id = request.Id,
            Name = request.Name,
            Version = request.Version,
            Image = request.Image,
            Status = request.Status,
        };
    }
}
